<?xml version="1.0" encoding="iso-8859-1"?>
<!DOCTYPE html 
     PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
     "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html>
<head>
  <title>exception_log (Daemons::Application)</title>
  <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1" />
  <link rel="stylesheet" href="../../.././rdoc-style.css" type="text/css" media="screen" />
</head>
<body class="standalone-code">
  <pre><span class="ruby-comment cmt"># File lib/daemons/application.rb, line 338</span>
    <span class="ruby-keyword kw">def</span> <span class="ruby-identifier">exception_log</span>
      <span class="ruby-keyword kw">return</span> <span class="ruby-keyword kw">unless</span> <span class="ruby-identifier">logfile</span>
      
      <span class="ruby-identifier">require</span> <span class="ruby-value str">'logger'</span>
      
      <span class="ruby-identifier">l_file</span> = <span class="ruby-constant">Logger</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">logfile</span>)
      
      <span class="ruby-comment cmt"># the code below finds the last exception</span>
      <span class="ruby-identifier">e</span> = <span class="ruby-keyword kw">nil</span>
      
      <span class="ruby-constant">ObjectSpace</span>.<span class="ruby-identifier">each_object</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">o</span><span class="ruby-operator">|</span>
        <span class="ruby-keyword kw">if</span> <span class="ruby-operator">::</span><span class="ruby-constant">Exception</span> <span class="ruby-operator">===</span> <span class="ruby-identifier">o</span>
          <span class="ruby-identifier">e</span> = <span class="ruby-identifier">o</span>
        <span class="ruby-keyword kw">end</span>
      }
     
      <span class="ruby-identifier">l_file</span>.<span class="ruby-identifier">info</span> <span class="ruby-value str">&quot;*** below you find the most recent exception thrown, this will be likely (but not certainly) the exception that made the application exit abnormally ***&quot;</span>
      <span class="ruby-identifier">l_file</span>.<span class="ruby-identifier">error</span> <span class="ruby-identifier">e</span>
      
      <span class="ruby-identifier">l_file</span>.<span class="ruby-identifier">info</span> <span class="ruby-value str">&quot;*** below you find all exception objects found in memory, some of them may have been thrown in your application, others may just be in memory because they are standard exceptions ***&quot;</span>
      
      <span class="ruby-comment cmt"># this code logs every exception found in memory</span>
      <span class="ruby-constant">ObjectSpace</span>.<span class="ruby-identifier">each_object</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">o</span><span class="ruby-operator">|</span>
        <span class="ruby-keyword kw">if</span> <span class="ruby-operator">::</span><span class="ruby-constant">Exception</span> <span class="ruby-operator">===</span> <span class="ruby-identifier">o</span>
          <span class="ruby-identifier">l_file</span>.<span class="ruby-identifier">error</span> <span class="ruby-identifier">o</span>
        <span class="ruby-keyword kw">end</span>
      }
      
      <span class="ruby-identifier">l_file</span>.<span class="ruby-identifier">close</span>
    <span class="ruby-keyword kw">end</span></pre>
</body>
</html>